### ISSUE TITLE
Adding volumes to nodepools on TKGS cluster results in 0 capacity

### ISSUE THREAD

**Issue by doddatpivotal** [vmware/terraform-provider-tanzu-mission-control#52]
### Describe the bug

I'm attempting to add an volume to my default node pool.  See capacity=50 below.  However, the TKC that is created in TKGS has a capacity value of 0, which the WCPMachine rejects.

TKC
```
    ...
    nodePools:
    - name: default-nodepool
      replicas: 1
      storageClass: vc01cl01-t0compute
      tkr:
        reference:
          name: v1.21.6---vmware.1-tkg.1.b3d708a
      vmClass: custom-worker-standard
      volumes:
      - capacity:
          storage: "0" # <= SEE BAD VALUE
        mountPath: /var/lib/containerd
        name: containerd
        storageClass: vc01cl01-t0compute
     ...
```

Teraform Resource
```
resource "tanzu-mission-control_cluster" "sandbox_1" {

  # only provision if sandbox_1 is true in tfvars
  count = var.sandbox_1 ? 1 : 0

  management_cluster_name = var.management_cluster
  provisioner_name        = "sandbox-vns"
  name                    = "sandbox-1"

  meta {
    labels = { "infra" : var.infra_name }
  }

  spec {
    cluster_group = tanzu-mission-control_cluster_group.sandbox.name
    tkg_service_vsphere {
      settings {
        network {
          pods {
            cidr_blocks = [
              "172.20.0.0/16", # pods cidr block by default has the value `172.20.0.0/16`
            ]
          }
          services {
            cidr_blocks = [
              "10.96.0.0/16", # services cidr block by default has the value `10.96.0.0/16`
            ]
          }
        }
        storage {
          classes = [
            var.storage_class,
          ]
          default_class = var.storage_class
        }
      }

      distribution {
        version = var.k8s_version
      }

      topology {
        control_plane {
          class             = "custom-cp-guaranteed"
          storage_class     = var.storage_class
          high_availability = false
        }
        node_pools {
          spec {
            worker_node_count = "1"

            cloud_label = {}
            node_label  = {}

            tkg_service_vsphere {
              class         = "custom-worker-standard"
              storage_class = var.storage_class
              volumes {
                capacity          = 50
                mount_path        = "/var/lib/containerd"
                name              = "containerd"
                pvc_storage_class = var.storage_class
              }
            }
          }
          info {
            name = "default-nodepool"
          }
        }
      }
    }
  }

  ready_wait_timeout = "15m"

}
```

### Reproduction steps

```bash
Simply terraform apply the resource above
```

### Expected behavior

The TKC should have the capacity of 50 applied to the volume, not 0

**Comment by shreyassreenivas** [comment]
Hi @doddatpivotal , Thank you for raising this bug and providing all the necessary details.
This issue was identified in our dev testing and the bug which you have reported has already been fixed in https://github.com/vmware/terraform-provider-tanzu-mission-control/pull/48

It was a tricky bug to root cause. The float values of the schema as not being read properly. We have addressed this issue and also ensured that such kind of data type conversion issue will not occur again by creating a library to read primitive types from schema.
